src/instructlab/data/generate.py

```diff
@@ -155,43 +156,71 @@ def generate(
     from instructlab.sdg.utils import GenerateException
 
     # First Party
-    from instructlab.model.backends.llama import ensure_server
+    from instructlab.model.backends.llama_cpp import ensure_server
 
     server_process = None
     server_queue = None
     prompt_file_path = config.DEFAULT_PROMPT_FILE
     if ctx.obj is not None:
         prompt_file_path = ctx.obj.config.generate.prompt_file
 
+    backend_instance = None
     if endpoint_url:
         api_base = endpoint_url
     else:
-        # Third Party
-        import llama_cpp
+        # First Party
+        from instructlab.model.backends import backends, llama_cpp, vllm
 
-        if not llama_cpp.llama_supports_gpu_offload():
-            # TODO: check for working offloading. The function only checks
-            # for compile time defines like `GGML_USE_CUDA`.
-            click.secho(
-                "llama_cpp_python is built without hardware acceleration. "
-                "ilab generate will be very slow.",
-                fg="red",
+        model_path = ctx.obj.config.serve.model_path
+        model_path = pathlib.Path(model_path)
+        backend = ctx.obj.config.serve.backend
+        try:
+            backend = backends.get(logger, model_path, backend)
+        except ValueError as e:
+            click.secho(f"Failed to determine backend: {e}", fg="red")
+            raise click.exceptions.Exit(1)
+
+        host = ctx.obj.config.serve.host_port.split(":")[0]
+        port = int(ctx.obj.config.serve.host_port.split(":")[1])
+
+        if backend == backends.LLAMA_CPP:
+            # Instantiate the llama server
```

Comment:
nit
```suggestion
            # Initialize the llama server
```

Reply:
It's effectively "instantiating" `backend_instance` as an object.